### PR TITLE
Domains: hide renew actions in purchase page for domains with pending registration status

### PR DESCRIPTION
**Pull request by gius80**
## Proposed Changes

For some tlds the registration is asynchronous and has to be confirmed by the registry. In case of .com.br tld, this process can take 3 days. We want to disallow domain renewal during this time window.

## Testing Instructions

Register a `.com.br` domain (or hack the `Domain_Management_Domain` class to set `pending_registration_at_registry=true`).

Visite the purchase detail page and verify that all the renew action are not visibile (see screenshot below).

### Before
domain-renewal-before

### After
domain-renewal-after

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/me/purchases/manage-purchase/index.tsx [hunk 1]
```diff
@@ -327,6 +327,16 @@ class ManagePurchase extends Component<
...
 	}
 
+	isPendingDomainRegistration( purchase: Purchase ): boolean {
+		if ( isDomainRegistration( purchase ) ) {
+			const domain = this.props.domainsDetails[ purchase.siteId ].find(
+				( domain ) => domain.name === purchase.meta
+			);
+			return domain?.pendingRegistrationAtRegistry ?? false;
+		}
+		return false;
```
Comment (leonardost) [inline client/me/purchases/manage-purchase/index.tsx]:
```suggestion
		if ( ! isDomainRegistration( purchase ) ) {
			return false;
		}

		const domain = this.props.domainsDetails[ purchase.siteId ].find(
			( domain ) => domain.name === purchase.meta
		);
		return domain?.pendingRegistrationAtRegistry ?? false;
```
Just a small suggestion 🙂 I know this is a tiny method and it doesn't make much difference, but I think early returns are more used in our codebase.

File: client/me/purchases/manage-purchase/test/purchase-management-buttons.js [hunk 2]
```diff
@@ -199,6 +199,38 @@ describe( 'Purchase Management Buttons', () => {
 		expect( screen.queryByText( /Cancel/ ) ).not.toBeInTheDocument();
 	} );
 
+	it( "don't renders renew buttons for domain with pending registration at registry", async () => {
```
Comment (leonardost) [inline client/me/purchases/manage-purchase/test/purchase-management-buttons.js]:
```suggestion
	it( "doesn't render renew buttons for domain with pending registration at registry", async () => {
```
Nit

File: client/me/purchases/manage-purchase/test/purchase-management-buttons.js [hunk 3]
```diff
@@ -199,6 +199,38 @@ describe( 'Purchase Management Buttons', () => {
...
+					<ManagePurchase
+						purchaseId={ Number( purchase.ID ) }
+						isSiteLevel
+						siteSlug="onecooltestsite.com"
+					/>
+				</ReduxProvider>
+			</QueryClientProvider>
+		);
+		expect( screen.queryByText( /Renew now/ ) ).not.toBeInTheDocument();
+	} );
```
Comment (leonardost) [inline client/me/purchases/manage-purchase/test/purchase-management-buttons.js:232]: Thanks for adding unit tests ❤️

**Review by leonardost — APPROVED** [review]
Tested and LGTM, thanks for this fix!

File: client/me/purchases/manage-purchase/index.tsx (hunk 1 above)
Comment (gius80) [reply client/me/purchases/manage-purchase/index.tsx]: Definitely better! Thanks for the suggestion! :)

**Pull request merged**